scripts/determine-runc-version

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env sh
+
+#   Copyright 2018-2020 Docker Inc.
+
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+
+#       http://www.apache.org/licenses/LICENSE-2.0
+
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+# Select the default version of runc based on the containerd source if no
+# RUNC_REF is set manually. For containerd > 1.5, the runc version commit/tag
+# is defined in go.mod. For older versions, use vendor.conf.
+#
+# See the install-runc script in the containerd repository:
+# https://github.com/containerd/containerd/blob/v1.5.0-beta.3/script/setup/install-runc#L24
+# https://github.com/containerd/containerd/blob/v1.4.0/script/setup/install-runc#L24
+runc_version() {
+	if [ -n "${RUNC_REF}" ]; then
+		echo "${RUNC_REF}"
+		return
+	fi
+
+	# shellcheck disable=SC2164
+	repo_abspath="$(cd -- "$(dirname "$0")/.." >/dev/null 2>&1 ; pwd -P)"
```

Comment:
don't know if there's a cleaner way; this variant seems to be portable (as in; it works on macOS as well, which doesn't support `readlink -f`)

Reply:
Just curiosity about `--`.  look likes `repo_abspath="$(cd "$(dirname "$0")/.." >/dev/null 2>&1 ; pwd -P)"` can work.

Reply:
Good point; TBH I was testing some variations that didn't work on macOS, and found this one, which did work, but was also wondering why the `--` was there, and forgot to test without; let me try

Reply:
Looks like it's the preferred way to do this (to be most portable); there's some discussion in https://go-review.googlesource.com/c/sys/+/210157/, which at some point suggests adding the `--` instead of skipping it.